samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -53,9 +53,11 @@
   StreamEdge(StreamSpec streamSpec, boolean isIntermediate, boolean isBroadcast, Config config) {
     this.streamSpec = streamSpec;
     this.isIntermediate = isIntermediate;
-    this.isBroadcast = isBroadcast;
+    // broadcast can be configured either by an operator or via the configs
+    this.isBroadcast =
+          isBroadcast || (config == null) ? false : new StreamConfig(config).getBroadcastEnabled(streamSpec.toSystemStream());
```

Comment:
Why would config be null here? I'd also prefer to extract variables for clarify instead of a ternary operator inside an or. It would also be clearer to make the caller do if possible. Otherwise the param isBroadcast doesn't mean what it says it means.

Reply:
I was trying to address some tests, that pass null. I agree it is not a valid case, so I modified the tests. I will remove this condition in PR846